kernel/threads-asm.s

```diff
@@ -42,13 +42,12 @@ schedule:
 .GLOBAL fork
 fork:
 	call fork_entry_point
```

Comment:
Why do we need fork and fork_entry_point?

Reply:
This is all wrapped up in the 'the fork stack must look a particular way' problem; I /think/ it isn't actually required here, but I'd want to test it pretty carefully... and I get rid of all of these functions three CLs later.